### PR TITLE
Add Install/Deprovision clusters with SiteConfig operator

### PR DESCRIPTION
**Pull request by amolnar-gh**
Issues:

- https://issues.redhat.com/browse/TELCODOCS-1875
- https://issues.redhat.com/browse/ACM-14071

### CODE REVIEW

File: acm_integration/siteconfig/deprovision_clusters.adoc [hunk 1]
```diff
@@ -0,0 +1,39 @@
+[#deprovision-clusters]
+= Deprovisioning {sno} clusters with the {sco}
+
+Delete your clusters with the {sco}.
```
Comment (swopebe) [inline acm_integration/siteconfig/deprovision_clusters.adoc]: We need more of a short description here. Why does the user want to do this? I am trying to get our short descriptions better across the board.

https://redhat-documentation.github.io/supplementary-style-guide/#shortdesc

File: acm_integration/siteconfig/deprovision_clusters.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+*Required access:* Cluster administrator
+
+[#install-clusters-preq]
+== Prerequisites
+
+* You have deployed your clusters with the {sco} by using the default cluster templates.
```
Comment (swopebe) [inline acm_integration/siteconfig/deprovision_clusters.adoc]: Can we make this more active:

You deployed your cluster...

File: acm_integration/siteconfig/deprovision_clusters.adoc [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+----
+$ oc get clusterinstance <cluster_name> -n <target_namespace>
+----
+
+See the following example output:
+
+[source,terminal]
+----
+Error from server (NotFound): clusterinstances.siteconfig.open-cluster-management.io "<cluster_name>" not found
+----
```
Comment (swopebe) [inline acm_integration/siteconfig/deprovision_clusters.adoc]: So when I first read this, I was like: Wait, I have an error.

Recommending:

See the following example output where the `(NotFound)` error indicates that your cluster is deprovisioned.

File: acm_integration/siteconfig/install_clusters.adoc [hunk 4]
```diff
@@ -0,0 +1,259 @@
+[#install-clusters]
+= Installing {sno} clusters with the {sco}
```
Comment (swopebe) [inline acm_integration/siteconfig/install_clusters.adoc]: I will pick up here tomorrow!

File: acm_integration/siteconfig/deprovision_clusters.adoc (hunk 2 above)
Comment (swopebe) [reply acm_integration/siteconfig/deprovision_clusters.adoc]: I have changed these a bit

File: acm_integration/siteconfig/install_clusters.adoc [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+
+Install your clusters with the {sco} by using the default cluster templates. Use the cluster templates for the Image-Based Install Operator to complete the procedure.
+
+*Required access:* Cluster administrator
+
+[#install-clusters-preq]
+== Prerequisites
+
```
Comment (swopebe) [inline acm_integration/siteconfig/install_clusters.adoc]: see comments here

File: acm_mce_integration/siteconfig/install_clusters.adoc [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+----
+$ oc apply -f enable-crun.yaml
+----
+
+[#install-clusters-templates]
+== Creating the cluster and node templates
+
+You can use the default cluster and node templates to install your cluster with the Image-based Install Operator. Complete the following steps to create the templates:
+
+. Create your cluster and node templates for the Image-based Install Operator
```
Comment (mpmaruthu) [inline acm_mce_integration/siteconfig/install_clusters.adoc:161]: In my opinion, this section doesn't require to publish as ibi cluster (both cluster + node level) templates  become available in `rhacm` namespace after enabling siteconfig operator.

sample refs:-

> [kni@registry.kni-qe-59 ~]$ oc -n rhacm get cm
NAME                                  DATA   AGE
ai-cluster-templates-v1               5      6d23h
ai-custom-cluster-templates-v1        5      3h22m
ai-custom-helix54-node-templates-v1   2      3h15m
ai-custom-helix77-node-templates-v1   2      3h15m
ai-node-templates-v1                  2      6d23h
console-config                        3      7d
grc-ca-bundle                         1      7d
ibi-cluster-templates-v1              3      6d23h
ibi-node-templates-v1                 3      6d23h
kube-root-ca.crt                      1      7d
manager-config                        1      7d
mch-image-manifest-2.12.0             49     7d
openshift-service-ca.crt              1      7d
search-ca-crt                         1      7d
search-indexer                        7      7d
search-postgres                       3      7d
search-v2-operator-manager-config     1      7d
trusted-ca-bundle                     1      7d
[kni@registry.kni-qe-59 ~]$

File: acm_mce_integration/siteconfig/install_clusters.adoc [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+  holdInstallation: false
+  extraManifestsRefs: <2>
+    - name: extra-machine-configs
+    - name: enable-crun
+  pullSecretRef:
+    name: "pull-secret" <3>
+  [...]
+  templateRefs: <4>
+    - name: ibi-cluster-templates-v1
+      namespace: siteconfig-operator
```
Comment (mpmaruthu) [inline acm_mce_integration/siteconfig/install_clusters.adoc:204]: namespace: **rhacm**

File: acm_mce_integration/siteconfig/install_clusters.adoc [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+  [...]
+  nodes:
+      [...]
+      bmcCredentialsName: <5>
+        name: "example-bmh-secret"
+      [...]
+      templateRefs: <6>
+        - name: ibi-node-templates-v1
+          namespace: siteconfig-operator
```
Comment (mpmaruthu) [inline acm_mce_integration/siteconfig/install_clusters.adoc:213]: namespace: **rhacm**

**Review by mpmaruthu — COMMENTED** [review]
Telco RAN QE: minor comments otherwise looks good to me.
/Peri M

**Pull request merged**